src/jit/regalloc.cpp

```diff
@@ -6438,15 +6438,21 @@ void               Compiler::rpPredictRegUse()
         // it must not be in a register trashed by the callee
         if (info.compCallUnmanaged != 0)
         {
-            LclVarDsc *     pinvokeVarDsc = &lvaTable[info.compLvFrameListRoot];
-
-            if (pinvokeVarDsc->lvTracked)
+            assert(!opts.ShouldUsePInvokeHelpers() || info.compLvFrameListRoot == BAD_VAR_NUM);
+            if (!opts.ShouldUsePInvokeHelpers())
             {
-                rpRecordRegIntf(RBM_CALLEE_TRASH, VarSetOps::MakeSingleton(this, pinvokeVarDsc->lvVarIndex)
-                                DEBUGARG("compLvFrameListRoot"));
+                noway_assert(info.compLvFrameListRoot < lvaCount);
+
+                LclVarDsc *     pinvokeVarDsc = &lvaTable[info.compLvFrameListRoot];
 
-                // We would prefer to have this be enregister in the PINVOKE_TCB register
-                pinvokeVarDsc->addPrefReg(RBM_PINVOKE_TCB, this);
+                if (pinvokeVarDsc->lvTracked)
+                {
+                    rpRecordRegIntf(RBM_CALLEE_TRASH, VarSetOps::MakeSingleton(this, pinvokeVarDsc->lvVarIndex)
+                                    DEBUGARG("compLvFrameListRoot"));
+
+                    // We would prefer to have this be enregister in the PINVOKE_TCB register
+                    pinvokeVarDsc->addPrefReg(RBM_PINVOKE_TCB, this);
+                }
```

Comment:
It would be nice to at least leave assert here - if somebody tries to use the legacy JIT for CoreRT.